src/controllers/auth.ts

```diff
@@ -0,0 +1,108 @@
+// Dependencies
+import UUID from 'uuid/v4'
+
+// Local Dependencies
+import { Request, Response } from '../interfaces'
+import { isSignatureValid, generateJwtToken } from '../helpers'
+import { models } from '../db'
+
+const signup = async (req: Request, res: Response, next: Function) => {
```

Comment:
In a future PR, we should probably have a check against a live data market to see if this user is on an approved list.

Reply:
Yeah, we'll still need to integrate this with blockchain once we nail down what we want that interaction to look like.